olp-cpp-sdk-dataservice-read/src/repositories/PartitionsRepository.cpp

```diff
@@ -63,6 +68,63 @@ TtlForLayer(const std::vector<olp::dataservice::read::model::Layer>& layers,
   return olp::client::ApiError(olp::client::ErrorCode::NotFound,
                                "Layer specified doesn't exist");
 }
+
+read::PartitionsResponse FindPartition(const read::QuadTreeIndex& quad_tree,
+                                       const read::TileRequest& request,
+                                       bool aggregated) {
+  const auto& tile_key = request.GetTileKey();
+
+  // Look for requested tile in the quad tree or its closest ancestor
+  const auto found_index_data = quad_tree.Find(tile_key, aggregated);
+
+  if (!found_index_data) {
+    OLP_SDK_LOG_ERROR_F(
```

Comment:
Error is a serious failure. I guess finding a tile in the quad tree does not qualify as a serious failure. A warning should be enough.